shape-detection/resources/endtoend-helpers.js

```diff
@@ -0,0 +1,90 @@
+function imageLoadedPromise(image) {
+    return new Promise(function(resolve, reject) {
+        if (image.complete)
+            resolve();
+        image.addEventListener("load", function() {
+            resolve();
+        });
+    });
+}
+
+function videoLoadedPromise(video) {
+    return new Promise(function(resolve, reject) {
+        video.addEventListener("loadeddata", function() {
+            resolve();
+        });
```

Comment:
Plus maybe want to make use of the `reject` in case the video fails loading. 

You might also just want to check the ready state of the video, and resolve early if it's already loaded? Might be overkill, but might catch annoying edge cases.

Reply:
I couldn't figure out how to do that. The MDN docs don't list any `ready` attribute or anything like that.